app/views/routes.py

```diff
@@ -0,0 +1,241 @@
+def create_incident():
+  data = request.get_json(force=True)
+
+  created_by = data.get('created_by', None)
+  incident_type = data.get('incident_type', None)
+  location = data.get('location', None)
+  phone_number = data.get('phone_number', None)
+  status  = 'draft'
+  images = data.get('images', None)
+  videos = data.get('videos', None)
+  comment = data.get('comment', None)
+
+  incident = Incident(created_by, incident_type, location, status, images, videos, comment)
+  respo = validate_incident(incident)
+
+  if respo:
+    return jsonify(respo), 400
+
+  incident_list.append(incident)
+
+  response = {
+    "status": 201,
+    "message": "Incident record created successfully ",
+    "data": [incident.to_dict()]
+  }
+  
+  return jsonify(response), 201
+
+def validate_incident(incident):
+  errors = []
+  user_exists = False
+
+  if incident:
+    created_by = incident.created_by
+    incident_type = incident.incident_type
+    location = incident.location
+    status = incident.status
+    images = incident.images
+    videos = incident.videos
+
+    for user in user_list:
+      if created_by == user.username:
+        user_exists = True
+      break
+
+    if not user_exists:
+      errors.append({
+        "user_error": "User assigned doesnot exist"
+      })
+
+    if not location:
+      errors.append({
+        "location_error": "Location is missing"
+      })
+
+    if not created_by:
+      errors.append({
+        "error_created": "Please provide valid fields"
+      })
+
+    if not incident_type:
+      error.append({		
+        "incident_type": "Please specify the incident type"
+      })
+
+  else:
+    errors.append({
+      "error": "Please provide incident details"
+    })
+
+  if len(errors) > 0:
+    return {
+      "status": 400,
+      "errors": errors
+    }
+  else:
+   return None
+
+@redflag.route('/incident', methods=['GET'])
+def fetch_all_incident():
+  if incident_list:
+    return jsonify({
+        "status": 200,
+        "data": [incident.to_dict() for incident in incident_list]
+      })
+  return jsonify({
+      "status":400,
+      "message": "No incidents created"
+      })
+
+@redflag.route('/incident/<incident_id>', methods=['GET'])
+def get_specific(incident_id):
+  for incident in incident_list:
+    if incident.id == incident_id:
+      return jsonify({
+        "status": 200,
+        "data": incident.to_dict()
+      })
+
+  return jsonify({
+    "status": 400,
+    "error": "Incident not found" 
+    })
+
+@redflag.route('/incident/<incident_id>/location', methods=['PATCH'])
+def update_location(incident_id):
+  data = request.get_json(force=True)
+  location = data.get('location', None)
+
+  if not location:
+    return jsonify({
+      "status": 400,
+      "message": "Please include Location of the record"
+    })
+
+  for incident in incident_list:
+    if incident.id == incident_id:
+      incident.location = location
+      return jsonify({
+        "status": 202,
```

Comment:
A status code of 202 means the server received the request but not yet acted on it, the best status code for this operation would be 200. The same applies to line 219 and 235.